### PR TITLE
#615 Fixed action image alignment issue in properties

### PR DESCRIPTION
**Pull request by nmgokhale**
Fixes https://github.com/elyra-ai/canvas/issues/615

To test this PR, I used this paramDef - 
[action_image_alignment_paramDef.txt](https://github.com/elyra-ai/canvas/files/6262643/action_image_alignment_paramDef.txt)

Screen Shot 2021-04-05 at 11 44 11 PM
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-properties/controls/control-factory.js [hunk 1]
```diff
@@ -332,9 +336,14 @@ export default class ControlFactory {
 			const className = control.className ? control.className : "";
 			return (
 				<div key={"properties-ctrl-" + control.name} data-id={"properties-ctrl-" + control.name}
-					className={classNames("properties-ctrl-wrapper", { "hide": hidden }, className)}
+					className={classNames(
+						"properties-ctrl-wrapper",
+						{ "hide": hidden, "action": !isNil(action) && has(control, "action.image.placement") },
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/control-factory.js:341]: Do you need to add this check `!isNil(action)`?  I would think the `has` check would be enough.

File: canvas_modules/common-canvas/src/common-properties/components/control-item/control-item.jsx [hunk 2]
```diff
@@ -110,7 +110,6 @@ class ControlItem extends React.Component {
 				>
 					{label}
 					{description}
-					{action}
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/components/control-item/control-item.jsx:113]: Isn't there more action code here that can be removed?

**Review by matthoward366 — COMMENTED** [review]
Are there any test files that have cases where the button is the left and right of the control?

**Comment by nmgokhale** [comment]
> Are there any test files that have cases where the button is the left and right of the control?

No. As per [documentation](https://github.com/elyra-ai/canvas/wiki/3.2-Common-Properties--UI-Hints#action-info), placement attribute is added under `image`. Even if I add `"placement": "right",` for button, there is no code in [actions/button/button.jsx](https://github.com/elyra-ai/canvas/blob/master/canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx#L45) to handle left or right placement.

**Comment by matthoward366** [comment]
How about image?  That's really what I meant above :)

**Comment by nmgokhale** [comment]
Test files - 
[action_image_alignment_paramDef.txt](https://github.com/elyra-ai/canvas/files/6266311/action_image_alignment_paramDef.txt)
[action_paramDef.txt](https://github.com/elyra-ai/canvas/files/6266317/action_paramDef.txt)

File: canvas_modules/common-canvas/src/common-properties/actions/image/image.scss [hunk 3]
```diff
@@ -20,12 +20,9 @@
 .properties-action-image  {
 	padding: 1px 2px;
 	cursor: pointer;
-	&.right {
-		margin-left: auto;
-		float: right;
-	}
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/src/common-properties/actions/image/image.scss:26]: I modified this CSS to work with new code for accessible controls.
For non-accessible controls, added CSS in `control-item.scss`.

**Review by matthoward366 — APPROVED** [review]
Looks good.

**Pull request merged**